src/UI/Component/Chart/Gauge/Factory.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Chart\Gauge;
+
+/**
+ * Interface Factory
+ * @package ILIAS\UI\Component\Chart\Gauge
+ */
+interface Factory {
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The standard gauge is usually the tool of choice. It is to be used, if the speedo needs to adapt
+     *     it's size depending on the screen size due to a more limited amount of space.
+     *   composition: >
+     *     The standard gauge is composed of one bar representing a value achieved in relation to a maximum
+     *     and a required value indicated by some pointer. The comparision value is represented by a second
+     *     bar below the first one. Also the percentage values of main and required are shown as text.
+     *   effect: >
+     *     On changing screen size they decrease their size including font size in various steps.
+     *
+     * rules:
+     *   composition:
+     *     1: Standard Gauges MAY contain a comparision value. It MUST be a numeric value between 0 and the maximum. It is represented as the second bar.
+     *     2: Standard Gauges MAY contain a main value text.
+     *     3: Standard Gauges MAY contain a required value text.
+     * ----
+     * @param int|float $maximum          Maximum reachable value.
+     * @param int|float $main             Main value to be displayed by main bar.
+     * @param int|float|null $required    Required value to be reached by main value.
+     * @param int|float|null $comparision Comparision value to be displayed by second bar.
+     * @return \ILIAS\UI\Component\Chart\Gauge\Standard
+     */
+    public function standard($maximum, $main, $required = null, $comparision = null);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The fixedSize gauge is mostly identical with the standard. It is used if
+     *     the gauge is to be rendered in it's original size, even on mobile devices.
+     *   composition: >
+     *     See composition description for standard gauge.
+     *
+     * rules:
+     *   composition:
+     *     1: See composition rules for standard gauge.
+     * ----
+     * @param int|float $maximum          Maximum reachable value.
+     * @param int|float $main             Main value to be displayed by main bar.
+     * @param int|float|null $required    Required value to be reached by main value.
+     * @param int|float|null $comparision Comparision value to be displayed by second bar.
+     * @return \ILIAS\UI\Component\Chart\Gauge\FixedSize
+     */
+    public function fixedSize($maximum, $main, $required = null, $comparision = null);
+
+    /**
+     * ---
+     * description:
+     *   purpose: >
+     *     The mini gauge is used, if it needs to be as small as possible, like in an heading. It

```

Comment:
The Mini Learning Progress Indicator is used to impart a reduced amount of information. The Mini Learning Progress Indicator is accompanies other structured data (i.e. heading or list item) and prevents informational overload from users.

Reply:
Im not sure about this formulation. The only information that is lost, is the output of text. The point of this element is that it will always be like a "miniature" of the component. Therefore I don't understand it as a reduction in functionality or information  but as a shrinked version.

Reply:
Then we do not need a Mini but just the Fixed Size set to very small.